src/main/kotlin/com/milaboratory/mixcr/cli/CommandAlign.kt

```diff
@@ -37,758 +35,743 @@ import com.milaboratory.core.sequence.NucleotideSequence
-import com.milaboratory.mixcr.basictypes.tag.TagTuple
-import com.milaboratory.mixcr.basictypes.tag.TagType
-import com.milaboratory.mixcr.basictypes.tag.TagValueType
-import com.milaboratory.mixcr.basictypes.tag.TagsInfo
-import com.milaboratory.mixcr.util.Defaults.default3
+import com.milaboratory.mixcr.basictypes.*
+import com.milaboratory.mixcr.basictypes.tag.*
 import com.milaboratory.mixcr.vdjaligners.VDJCAligner
 import com.milaboratory.mixcr.vdjaligners.VDJCAlignerParameters
 import com.milaboratory.mixcr.vdjaligners.VDJCAlignmentFailCause
 import com.milaboratory.mixcr.vdjaligners.VDJCAlignmentResult
-import com.milaboratory.mixcr.vdjaligners.VDJCParametersPresets
-import com.milaboratory.primitivio.buffered
-import com.milaboratory.primitivio.chunked
-import com.milaboratory.primitivio.forEach
-import com.milaboratory.primitivio.invoke
-import com.milaboratory.primitivio.map
-import com.milaboratory.primitivio.mapInParallel
-import com.milaboratory.primitivio.ordered
-import com.milaboratory.primitivio.unchunked
+import com.milaboratory.primitivio.*
 import com.milaboratory.util.CanReportProgress
-import com.milaboratory.util.GlobalObjectMappers
-import com.milaboratory.util.JsonOverrider
 import com.milaboratory.util.ReportHelper
 import com.milaboratory.util.ReportUtil
 import com.milaboratory.util.SmartProgressReporter
 import io.repseq.core.Chains
-import io.repseq.core.GeneFeature.VRegion
-import io.repseq.core.GeneFeature.VRegionWithP
-import io.repseq.core.GeneFeature.encode
+import io.repseq.core.GeneFeature.*
 import io.repseq.core.GeneType
 import io.repseq.core.VDJCLibrary
 import io.repseq.core.VDJCLibraryRegistry
-import picocli.CommandLine
-import picocli.CommandLine.Option
-import java.io.File
+import picocli.CommandLine.*
 import java.io.FileInputStream
-import java.io.IOException
-import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.Paths
 import java.util.*
 import java.util.regex.Pattern
 import java.util.stream.Collectors
+import kotlin.io.path.Path
+import kotlin.io.path.readText
 import kotlin.math.max
 
-@CommandLine.Command(
-    name = CommandAlign.ALIGN_COMMAND_NAME,
-    sortOptions = false,
-    separator = " ",
-    description = ["Builds alignments with V,D,J and C genes for input sequencing reads."]
-)
-class CommandAlign : MiXCRCommand() {
-    @CommandLine.Parameters(
-        arity = "2..3",
-        paramLabel = "files",
-        hideParamSyntax = true,
-        description = [
-            "file_R1.(fastq[.gz]|fasta|bam|sam) [file_R2.(fastq[.gz]|bam|sam)] [file_RN.(bam|sam)] alignments.vdjca",
-            "Use {{n}} if you want to concatenate files from multiple lanes, like:",
-            "my_file_L{{n}}_R1.fastq.gz my_file_L{{n}}_R2.fastq.gz"
-        ]
+object CommandAlign {
+    const val COMMAND_NAME = "align"
+
+    data class Params(
+        @JsonProperty("species") val species: String = "",
+        @JsonProperty("libraryName") val libraryName: String = "default",
```

Comment:
why do you set default value? this constructor never executed

Reply:
Yap. Will remove this.